sdks/python/apache_beam/utils/pipeline_options_test.py

```diff
@@ -170,6 +172,26 @@ def test_template_location(self):
     options = PipelineOptions(flags=[''])
     self.assertEqual(options.get_all_options()['template_location'], None)
 
+  def test_static_value_provider_of(self):
+    class TestOptions(PipelineOptions):
+      @classmethod
+      def _add_argparse_args(cls, parser):
+        parser.add_argument(
+            '--int_flag',
+            type=static_value_provider_of(int),
```

Comment:
How would the api look like with runtime value providers?

If users are going to mark an argument as `static_value_provider_of`, that means that they would know that this argument will be given at pipeline construction time. They might as well define it as a regular argument.

I would assume that a user will define type as a value_provider argument and based on that being available at pipeline construction time or not it will be converted to a static or dynamic value provider type. (@sammcveety to make sure that this API is consistent with Java SDK)